### PR TITLE
gio: add Windows stream types

### PR DESCRIPTION
**Pull request by pbor**

### CODE REVIEW

File: gio/src/win32_input_stream.rs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+impl fmt::Display for Win32InputStream {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.write_str("Win32InputStream")
+    }
+}
+
+impl Win32InputStream {
+    #[allow(clippy::missing_safety_doc)]
```
Comment (bilelmoussaoui) [inline gio/src/win32_input_stream.rs:43]: Could use a doc alias

Comment (GuillaumeGomez) [reply gio/src/win32_input_stream.rs:43]: *should

File: gio/src/win32_input_stream.rs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+use crate::InputStream;
+use crate::PollableInputStream;
+use glib::object::{Cast, IsA};
+use glib::translate::*;
+use std::fmt;
+
+use std::os::windows::io::{AsRawHandle, FromRawHandle, IntoRawHandle, RawHandle};
+
+glib::wrapper! {
+    pub struct Win32InputStream(Object<ffi::GWin32InputStream, ffi::GWin32InputStreamClass>) @extends InputStream, @implements PollableInputStream;
```
Comment (sdroege) [inline gio/src/win32_input_stream.rs:12]: It does not implement `PollableInputStream`

File: gio/src/win32_output_stream.rs [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+use crate::OutputStream;
+use crate::PollableOutputStream;
+use glib::object::{Cast, IsA};
+use glib::translate::*;
+use std::fmt;
+
+use std::os::windows::io::{AsRawHandle, FromRawHandle, IntoRawHandle, RawHandle};
+
+glib::wrapper! {
+    pub struct Win32OutputStream(Object<ffi::GWin32OutputStream, ffi::GWin32OutputStreamClass>) @extends OutputStream, @implements PollableOutputStream;
```
Comment (sdroege) [inline gio/src/win32_output_stream.rs:12]: It does not implement `PollableOutputStream`

**Review by sdroege — COMMENTED** [review]
Looks good to me otherwise. Did you confirm that it actually compiles and works on Windows?

**Comment by pbor** [comment]
> Looks good to me otherwise. Did you confirm that it actually compiles and works on Windows?

Yep. tested in the new CI :)

**Review by sdroege — APPROVED** [review]

**Pull request merged**